concordium-sdk-examples/src/main/java/com/concordium/sdk/examples/contractexample/cis2nft/Cis2NftParameters.java

```diff
@@ -0,0 +1,182 @@
+ */
+public class Cis2NftParameters {
+    private static final AccountAddress ACCOUNT_ADDRESS = AccountAddress.from("3XSLuJcXg6xEua6iBPnWacc3iWh93yEDMCqX8FbE3RDSbEnT9P");
+    private static final ContractAddress CONTRACT_ADDRESS_1 = ContractAddress.from(1, 0);
+    private static final ContractAddress CONTRACT_ADDRESS_2 = ContractAddress.from(2, 0);
+    private static final String CIS_2_NFT_CONTRACT_NAME = "cis2_nft";
+
+    private static final String SCHEMA_PATH = "./src/main/java/com/concordium/sdk/examples/contractexample/cis2nft/cis2-nft.schema.bin";
+    /**
+     * Generates and initializes {@link MintParams} for the 'mint' method of a cis2-nft contract.
+     *
+     * @return initialized {@link MintParams}.
+     */
+    @SneakyThrows
+    public static SchemaParameter generateMintParams() {
+        Schema schema = Schema.from(Files.readAllBytes(Paths.get(SCHEMA_PATH)), SchemaVersion.V3);
+        ReceiveName mintParamsReceiveName = ReceiveName.from(CIS_2_NFT_CONTRACT_NAME, "mint");
+        List<TokenIdU32> tokens = new ArrayList<>();
+        tokens.add(TokenIdU32.from(2));
+        tokens.add(TokenIdU32.from(22));
+        tokens.add(TokenIdU32.from(2132));
+        SchemaParameter mintParameter = new MintParams(schema, mintParamsReceiveName, ACCOUNT_ADDRESS, tokens);
+        mintParameter.initialize(true);
+        return mintParameter;
+    }
+
+    /**
+     * Generates and initializes {@link NftTransferParam} for the 'transfer' method of a cis2-nft contract.
+     *
+     * @return initialized {@link NftTransferParam}.
+     */
+    @SneakyThrows
+    public static SchemaParameter generateTransferParams() {
+        Schema schema = Schema.from(Files.readAllBytes(Paths.get(SCHEMA_PATH)), SchemaVersion.V3);
+        ReceiveName nftTransferReceiveName = ReceiveName.from(CIS_2_NFT_CONTRACT_NAME, "transfer");
+        TokenIdU32 tokenId = TokenIdU32.from(12);
+        TokenAmountU8 amount = TokenAmountU8.from(1);
+        AbstractAddress from = CONTRACT_ADDRESS_1;
+        Receiver to = new Receiver(CONTRACT_ADDRESS_2, "mint");
+        List<UInt8> data = new ArrayList<>();
+        data.add(UInt8.from(123));
+        data.add(UInt8.from(23));
+        NftTransfer transfer = new NftTransfer(tokenId, amount, from, to, data);
+        List<NftTransfer> transfers = new ArrayList<>();
+        transfers.add(transfer);
+        SchemaParameter transferParameter = new NftTransferParam(schema, nftTransferReceiveName, transfers);
+        transferParameter.initialize(true);
+        return transferParameter;
+    }
+
+    /**
+     * Generates and initializes {@link UpdateOperatorParams} for the 'updateOperator' method of a cis2-nft contract.
+     *
+     * @return initialized {@link UpdateOperatorParams}.
+     */
+    @SneakyThrows
+    public static SchemaParameter generateUpdateOperatorParams() {
+        Schema schema = Schema.from(Files.readAllBytes(Paths.get(SCHEMA_PATH)), SchemaVersion.V3);
```

Comment:
The version of the `Schema`. It is needed for the rust layer to serialize the parameters. I see you commented on it further below as well, should it just be V3 as default when initalizing `Schema`s without providing a specific version?.

Reply:
I am not a big fan of defaults like this, since they tend to lead to errors. But in this case a schema serialied as V1 or V2 will also parse as a V3 schema (with the same meaning) so it's kind of OK.